patches/0004-Output-to-hashed-mappings.patch

```diff
@@ -0,0 +1,1622 @@
++	public static IntermediateMappingsService create(IntermediateMappingsProvider intermediateMappingsProvider, MinecraftProvider minecraftProvider, MappingsNamespace intermediateMappings) {
+ 		final Path intermediaryTiny = minecraftProvider.file(intermediateMappingsProvider.getName() + ".tiny").toPath();
+ 
+ 		try {
+@@ -80,14 +82,14 @@ public final class IntermediateMappingsService implements SharedService {
+ 			throw new UncheckedIOException("Failed to provide intermediate mappings", e);
+ 		}
+ 
+-		return new IntermediateMappingsService(intermediaryTiny);
++		return new IntermediateMappingsService(intermediaryTiny, intermediateMappings);
+ 	}
+ 
+ 	private MemoryMappingTree createMemoryMappingTree() {
+ 		final MemoryMappingTree tree = new MemoryMappingTree();
+ 
+ 		try {
+-			MappingNsCompleter nsCompleter = new MappingNsCompleter(tree, Collections.singletonMap(MappingsNamespace.NAMED.toString(), MappingsNamespace.INTERMEDIARY.toString()), true);
++			MappingNsCompleter nsCompleter = new MappingNsCompleter(tree, Collections.singletonMap(MappingsNamespace.NAMED.toString(), intermediateMappings.toString()), true);
+ 
+ 			try (BufferedReader reader = Files.newBufferedReader(getIntermediaryTiny(), StandardCharsets.UTF_8)) {
+ 				Tiny2Reader.read(reader, nsCompleter);
+diff --git a/src/main/java/net/fabricmc/loom/configuration/providers/mappings/LayeredMappingSpecBuilderImpl.java b/src/main/java/net/fabricmc/loom/configuration/providers/mappings/LayeredMappingSpecBuilderImpl.java
+index 26dd21b910a8789d817ff4048aa0a0d5740f6327..07028b90cd9b41c276533fda7c6e1930b7204842 100644
+--- a/src/main/java/net/fabricmc/loom/configuration/providers/mappings/LayeredMappingSpecBuilderImpl.java
++++ b/src/main/java/net/fabricmc/loom/configuration/providers/mappings/LayeredMappingSpecBuilderImpl.java
+@@ -38,7 +38,7 @@ import net.fabricmc.loom.api.mappings.layered.spec.MojangMappingsSpecBuilder;
+ import net.fabricmc.loom.api.mappings.layered.spec.ParchmentMappingsSpecBuilder;
+ import net.fabricmc.loom.configuration.providers.mappings.extras.signatures.SignatureFixesSpec;
+ import net.fabricmc.loom.configuration.providers.mappings.file.FileMappingsSpecBuilderImpl;
+-import net.fabricmc.loom.configuration.providers.mappings.intermediary.IntermediaryMappingsSpec;
++import net.fabricmc.loom.configuration.providers.mappings.hashed.HashedMappingsSpec;
+ import net.fabricmc.loom.configuration.providers.mappings.mojmap.MojangMappingsSpecBuilderImpl;
+ import net.fabricmc.loom.configuration.providers.mappings.parchment.ParchmentMappingsSpecBuilderImpl;
+ 
+@@ -79,8 +79,8 @@ public class LayeredMappingSpecBuilderImpl implements LayeredMappingSpecBuilder
+ 
+ 	public LayeredMappingSpec build() {
+ 		List<MappingsSpec<?>> builtLayers = new LinkedList<>();
+-		// Intermediary is always the base layer
+-		builtLayers.add(new IntermediaryMappingsSpec());
++		// Hashed is always the base layer
++		builtLayers.add(new HashedMappingsSpec());
```

Comment:
This is okay as long as we're able to layer Intermediary on top of Hashed, for e.g. Quilted Fabric API

Reply:
Why would intermediary get layered on top of hashed? All the dependencies get remapped to hashed and intermediary is merged into the mapping tree regardless